kernel/comps/virtio/src/device/block/device.rs

```diff
@@ -406,6 +407,54 @@ impl DeviceInner {
         }
     }
 
+    /// Flushes any cached data from the guest to the persistent storage on the host.
+    /// This will be ignored if the device doesn't support the `VIRTIO_BLK_F_FLUSH` feature.
+    fn flush(&self, bio_request: BioRequest) {
+        if self.transport.lock().device_features() & BlockFeatures::FLUSH.bits() == 0 {
+            return;
```

Comment:
I think we should mark the bio request submitted and completed here to avoid keeping the user waiting.